src/views/Entries/QualityControl/index.tsx

```diff
@@ -229,17 +233,21 @@ function QualityControl(props: Props) {
         setDeletedEntries(oldDeletedEntries => ({ ...oldDeletedEntries, [entryId]: true }));
     }, [setDeletedEntries, getEntriesWithStats]);
 
-    const handleSelection = useCallback(value => (
-        selected && selected.id === value.id ?
-            setSelection({ matrixKey: undefined }) : setSelection({ matrixKey: value })
-    ), [selected, setSelection]);
+    const handleSelection = useCallback((value) => {
+        const isSelected = tocFilters.some(s => idSelector(s) === idSelector(value));
+        if (isSelected) {
```

Comment:
I think we should handle this from within the component

Reply:
yes